### PR TITLE
[HOPSWORKS-2636] Add option to specify Kubernetes node labels from cl…

### PR DESCRIPTION
**Pull request by SirOibaf**
…uster definition

### CODE REVIEW

File: recipes/addons.rb [hunk 1]
```diff
@@ -70,6 +70,34 @@
...
+    k8s_node node_name
+    action :taint
+  end
+end
+
+# Apply node labels
+node['kube-hops']['labels'].split(")").each do |node_label|
+  node_label = node_label[1, node_label.length-1]
+  node_name = node_label.split(",")[0]
+  label = node_label.split(",")[1]
```
Comment (javierdlrm) [inline recipes/addons.rb:91]: Just by curiosity, why not store the split in a variable?

**Review by javierdlrm — APPROVED** [review]

File: recipes/addons.rb (hunk 1 above)
Comment (SirOibaf) [reply recipes/addons.rb:91]: cleaned up.

File: metadata.rb [hunk 2]
```diff
@@ -168,3 +176,11 @@
 attribute "kube-hops/kfserving/img_tar_url",
           :description =>  "Remote container images registry from which to fetch the kfserving and dependencies images",
           :type => 'string'
+
+attribute "kube-hops/serving_node_labels",
+          :description =>  "The labels to apply to the KfServing pods",
```
Comment (javierdlrm) [inline metadata.rb:181]:
```suggestion
          :description =>  "The labels used for node selection in model serving pods, in the format key1=value1,key2=value2",
```

We also apply node selection when serving models with regular kubernetes deployments

File: metadata.rb [hunk 3]
```diff
@@ -168,3 +176,11 @@
...
+          :type => 'string'
+
+attribute "kube-hops/serving_node_tolerations",
+          :description =>  "The tolerations to apply to the KfServing pods",
```
Comment (javierdlrm) [inline metadata.rb:185]:
```suggestion
          :description =>  "The tolerations attached to model serving pods, in the format key1:operator1[:value1]:effect1,key2:operator2[:value2]:effect2",
```

Same.

File: templates/default/knative-serving.yml.erb [hunk 4]
```diff
@@ -601,10 +601,10 @@ data:
     kubernetes.podspec-affinity: "disabled"
 
     # Indicates whether Kubernetes nodeSelector support is enabled
-    kubernetes.podspec-nodeselector: "disabled"
+    kubernetes.podspec-nodeselector: "enabled"
 
     # Indicates whether Kubernetes tolerations support is enabled
-    kubernetes.podspec-tolerations: "disabled"
+    kubernetes.podspec-tolerations: "enabled"
```
Comment (javierdlrm) [inline templates/default/knative-serving.yml.erb:607]: This is the example configuration. Please add these lines immediately after `data:` (outside `_example:`) https://github.com/logicalclocks/kube-hops-chef/blob/24cb9e9cd56d8698d74e42f862ec242f597d2e00/templates/default/knative-serving.yml.erb#L580

**Review by javierdlrm — CHANGES_REQUESTED** [review]

**Review by javierdlrm — APPROVED** [review]

**Pull request merged**